harness/determined/pytorch/deepspeed/_mpu.py

```diff
@@ -0,0 +1,103 @@
+import abc
+from typing import cast
+
+from deepspeed.runtime.pipe import topology
+
+from determined import _core
+
+
+class ModelParallelUnit:
+    """
+    This class contains the functions we expect in order to accurately carry out parallel training.
+    For custom model parallel training, you need to subclass and override the functions before
+    passing it to the :class:`~determined.pytorch.deepspeed.DeepSpeedTrialContext` by calling
+    ``context.wrap_mpu(mpu)``.
+    """
+
+    @abc.abstractmethod
+    def __init__(self) -> None:
+        pass
```

Comment:
Don't make this part of the interface (leave it undefined here, with no abstractmethod)